jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/BuildImageStep.java

```diff
@@ -186,4 +189,59 @@ public DescriptorDigest getDiffId() throws LayerPropertyNotFoundException {
       return imageBuilder.build();
     }
   }
+
+  /**
+   * Computes the image entrypoint. If {@link ContainerConfiguration#getEntrypoint()} is null, the
+   * entrypoint is inherited from the base image. Otherwise {@link
+   * ContainerConfiguration#getEntrypoint()} is returned.
+   *
+   * @param baseImage the base image
+   * @param containerConfiguration the container configuration
+   * @return the container entrypoint
+   */
+  @Nullable
+  private ImmutableList<String> computeEntrypoint(
+      Image<Layer> baseImage, ContainerConfiguration containerConfiguration) {
+    if (baseImage.getEntrypoint() == null || containerConfiguration.getEntrypoint() != null) {
+      return containerConfiguration.getEntrypoint();
+    }
+
+    buildConfiguration
+        .getEventDispatcher()
+        .dispatch(
+            LogEvent.lifecycle(
+                "Container entrypoint set to "
+                    + baseImage.getEntrypoint()
+                    + " (inherited from base image)"));
+    return baseImage.getEntrypoint();
+  }
+
+  /**
+   * Computes the image program arguments. If {@link ContainerConfiguration#getEntrypoint()} and
+   * {@link ContainerConfiguration#getProgramArguments()} are null, the program arguments are
+   * inherited from the base image. Otherwise {@link ContainerConfiguration#getProgramArguments()}
+   * is returned.
+   *
+   * @param baseImage the base image
+   * @param containerConfiguration the container configuration
+   * @return the container program arguments
+   */
+  @Nullable
+  private ImmutableList<String> computeProgramArguments(
+      Image<Layer> baseImage, ContainerConfiguration containerConfiguration) {
+    if (baseImage.getProgramArguments() == null
+        || containerConfiguration.getEntrypoint() != null
+        || containerConfiguration.getProgramArguments() != null) {
```

Comment:
Looks like `ContainerConfiguration.getProgramArguments()` can never be null.

- `ContainerConfiguration.programArguments` is set to `JibContainerBuilder.programArguments` in `JibContainerBuilder.toContainerConfiguration()`.
- `JibContainerBuilder.programArguments` is in turn set to `jibExtension.getContainer().getArgs()` in `PluginConfigurationProcessor.processCommonConfiguration()` for Gradle.
- `jibExtension.getContainer().getArgs()` is not `@Nullable`, so never null.

That means, we will never inherit program arguments, right? I think `PluginConfigurationProcessor.processCommonConfiguration()` should set `JibContainerBuilder.programArguments` to `null` if WAR, similar to what this PR does with Entrypoint in `computeEntrypoint`.

Reply:
Wait, I do see the program arguments are being inherited in my actual small testing. Looking into what's happening.

Reply:
Sorry, never mind. I was looking at a wrong branch.